src/modules/commander/Commander.cpp

```diff
@@ -718,6 +718,30 @@ Commander::handle_command(const vehicle_command_s &cmd)
 		}
 		break;
 
+	case vehicle_command_s::VEHICLE_CMD_DO_CHANGE_ALTITUDE: {
+
+			// Just switch the flight mode here, the navigator takes care of
+			// doing something sensible with the coordinates. Its designed
+			// to not require navigator and command to receive / process
+			// the data at the exact same time.
+
+			if (_user_mode_intention.change(vehicle_status_s::NAVIGATION_STATE_AUTO_LOITER)) {
+				cmd_result = vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED;
+
+			} else {
+				cmd_result = vehicle_command_ack_s::VEHICLE_CMD_RESULT_TEMPORARILY_REJECTED;
+				mavlink_log_critical(&_mavlink_log_pub, "Change altitude command rejected\t");
+				/* EVENT
+					* @description Check for a valid position estimate
+					*/
+				events::send(events::ID("commander_change_altitude_rejected"),
+				{events::Log::Error, events::LogInternal::Info},
+				"Change altitude command rejected");
```

Comment:
You could simplify this to `printRejectMode(vehicle_status_s::NAVIGATION_STATE_AUTO_LOITER);` (the same for reposition)

Reply:
done